src/httpserver.cpp

```diff
@@ -586,11 +604,10 @@ void HTTPRequest::WriteReply(int nStatus, const std::string& strReply)
         // workaround above.
         if (event_get_version_number() >= 0x02010600 && event_get_version_number() < 0x02020001) {
             evhttp_connection* conn = evhttp_request_get_connection(req_copy);
-            if (conn) {
+            LOCK(g_http_connections_cs);
+            if (conn && g_http_connections.count(conn)) {
```

Comment:
Comment above "Re-enable reading..." is out of date, or at least incomplete now, and would be good to update.

I think I need to look more into the workaround here, but it's not clear to me why this no longer enables write events during shutdown (or what the effect of that is). Hopefully it doesn't mean that a pending writes would never complete...

Reply:
Writes are always enabled and there is no reason to enable reading if `g_http_connections.count(conn) == 0`.